homeassistant/components/sensor/statistics.py

```diff
@@ -183,36 +178,45 @@ def async_update(self):
         if self._max_age is not None:
             self._purge_old()
 
+        self.count = len(self.states)
+
         if not self.is_binary:
             try:  # require only one data point
                 self.mean = round(statistics.mean(self.states), 2)
                 self.median = round(statistics.median(self.states), 2)
             except statistics.StatisticsError as err:
-                _LOGGER.error(err)
+                _LOGGER.debug(err)
                 self.mean = self.median = STATE_UNKNOWN
 
             try:  # require at least two data points
                 self.stdev = round(statistics.stdev(self.states), 2)
                 self.variance = round(statistics.variance(self.states), 2)
             except statistics.StatisticsError as err:
-                _LOGGER.error(err)
+                _LOGGER.debug(err)
                 self.stdev = self.variance = STATE_UNKNOWN
 
             if self.states:
-                self.count = len(self.states)
                 self.total = round(sum(self.states), 2)
                 self.min = min(self.states)
                 self.max = max(self.states)
+                self.min_age = self.ages[0]
+                self.max_age = self.ages[-1]
                 self.change = self.states[-1] - self.states[0]
                 self.average_change = self.change
+                self.change_rate = 0
+
                 if len(self.states) > 1:
                     self.average_change /= len(self.states) - 1
-                if self._max_age is not None:
-                    self.max_age = max(self.ages)
-                    self.min_age = min(self.ages)
+
+                    time_diff = (self.max_age - self.min_age).total_seconds()
+                    if time_diff > 0:
+                        self.change_rate = self.average_change / time_diff
```

Comment:
This should be rounded.

Reply:
I think no values should be rounded at all as the decimal places depend on the original sensor value. The hard coded value 2 has bene chosen arbitrarily and for my precisely use case change_rate has at least 2 zero decimal places which would result in an always zero value.

These are 2 examples of my sensor that calculates the change rate of a water level based on a level of 100%:
```
average_change: -0.12000000000000455
change: -0.2400000000000091
change_rate: -0.004051726227830205
count: 3
max_age: 2018-08-10T03:50:05.168733+00:00
max_value: 99.84
min_age: 2018-08-10T03:49:35.551727+00:00
min_value: 99.6
sampling_size: 3
total: 299.2
unit_of_measurement: %

average_change: 0.11999999999999744
change: 0.23999999999999488
change_rate: 0.0013346913966593103
count: 3
max_age: 2018-08-10T03:52:15.271830+00:00
max_value: 99.44
min_age: 2018-08-10T03:50:45.363406+00:00
min_value: 99.2
sampling_size: 3
total: 298
unit_of_measurement: %
```
change_rate is then used to calculate a rate in L/min which results in a large enough value (above example ~0.9 L/min) but this would be to specific for a general sensor, so I chose seconds instead of minutes.

I don't use the other attributes but I imagine this could be the case for others too. So I would propose to remove the round for all values and just export the raw values.  Also most times you still have to use a template sensor to access the actual value, so everyone can round to his specific needs.

Reply:
If you are exposing a value to the frontend then the amount of decimal places must be limited. 10 and more decimal places will not make the result more accurate or better readable. We don't care about the amount of decimal places used internally but for the frontend.

You will always find uses cases where the amount of decimal places or the unit of measurement doesn't match your needs. For all those cases we have the `template` sensor and `custom_components` which let you do exactly what you want.